#11 add theme design-system

Homa-Shafiei opened the pull request.

Homa-Shafiei committed:
#11 feature: add theme design-system

viamgr commented on designsystem/src/main/kotlin/com/epiclabs/thinktok/designsystem/theme/Theme.kt:
```diff
@@ -0,0 +1,122 @@
+    if (view.isInEditMode.not()) {
+        SideEffect {
+            val window = (view.context as Activity).window
+            window.statusBarColor = colorScheme.primary.toArgb()
```
I think we can fix this warning: 'setter for statusBarColor: Int' is deprecated. Deprecated in Java

viamgr commented on designsystem/src/main/kotlin/com/epiclabs/thinktok/designsystem/theme/Theme.kt:
```diff
@@ -0,0 +1,122 @@
+            window.statusBarColor = colorScheme.primary.toArgb()
+            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = useDarkTheme
+            WindowCompat.getInsetsController(window, view).isAppearanceLightNavigationBars = useDarkTheme.not()
+            window.navigationBarColor = colorScheme.background.toArgb()
```
'setter for navigationBarColor: Int' is deprecated. Deprecated in Java

viamgr commented on designsystem/src/main/kotlin/com/epiclabs/thinktok/designsystem/theme/Type.kt:
```diff
@@ -0,0 +1,36 @@
+                lineHeight = 24.sp,
+                letterSpacing = 0.5.sp,
+            ),
+    /* Other default text styles to override
```
We don't need to override the default behavior and we can remove these bodyLarge and comments.

viamgr commented on settings.gradle.kts:
```diff
@@ -25,6 +25,7 @@ dependencyResolutionManagement {
 rootProject.name = "Think Tok"
 include(":app")
 include(":core")
+include(":designsystem")
```
Could you please add this project dependency to the main:ui module?

viamgr commented on app/build.gradle.kts:
```diff
@@ -10,6 +10,7 @@ android {
 
 dependencies {
     implementation(project(":core"))
+    implementation(project(":designsystem"))
```
I think the app module should not be dependent on the design system module. What do you think?

viamgr committed:
Refactored MainScreen location and updated the import in MainActivity

- Moved `MainScreen` from `com.epiclabs.thinktok.main.ui` to `com.epiclabs.thinktok.main.ui.main`.
- Updated the import statement in `MainActivity` to reflect the new location of `MainScreen`.

viamgr committed:
Refactor: Updated typography and theme in design system

- Updated `Typography` to use `TextStyle` constructor for `bodyLarge`.
- Refactored `ThinkTokTheme` to use if-else for `dynamicLightColorScheme` and `dynamicDarkColorScheme`.

viamgr committed:
Refactor: Updated typography and theme in design system

- Updated `Typography` to use `TextStyle` constructor for `bodyLarge`.
- Refactored `ThinkTokTheme` to use if-else for `dynamicLightColorScheme` and `dynamicDarkColorScheme`.

viamgr committed:
Merge remote-tracking branch 'origin/11-android-theme-design-system' into 11-android-theme-design-system

viamgr committed:
Fixed conflict issues

viamgr reviewed (approved):
Great Job
